include/aspect/assembly.h

```diff
@@ -119,6 +116,9 @@ namespace aspect
            */
           MaterialModel::MaterialModelInputs<dim> face_material_model_inputs;
           MaterialModel::MaterialModelOutputs<dim> face_material_model_outputs;
+
+          std::vector<double> mass_densities;
+          std::vector<double> adiabatic_density_gradients;
```

Comment:
Can you add a comment what these correspond to? I'm also slightly confused by the fact that the second variable is a *gradient* but has data type `double`. A comment would probably be helpful.